File: keeper/src/main/java/com/github/abhijitpparate/keeper/screen/home/presenter/HomeContract.java
package com.github.abhijitpparate.keeper.screen.home.presenter;


import android.support.annotation.StringRes;

import com.github.abhijitpparate.keeper.data.database.Note;
import com.github.abhijitpparate.keeper.data.database.Profile;

import java.util.List;

public interface HomeContract {

    public static enum NoteType {
        TEXT,
        IMAGE,
        VIDEO,
        AUDIO,
        LOCATION,
        DRAWING,
        LIST
    }

    interface View {
        void setUserInfo(Profile profile);
        void setNotes(List<Note> notes);

        void showLoginScreen();
        void showNewNoteScreen(NoteType noteType);

        void showNoteScreen(Note note);

        void setPresenter(Presenter presenter);
        void showProgressBar(boolean bool);
        void makeToast(@StringRes int strId);
        void makeToast(String message);
    }

    interface Presenter {

        void onRefresh();

        void onLogoutClick();
        void onNewNoteClick(NoteType noteType);

        void onNoteClick(Note note);

        void subscribe();
        void unsubscribe();
    }
}

File: keeper/src/main/java/com/github/abhijitpparate/keeper/screen/home/presenter/HomeContract.kt
package com.github.abhijitpparate.keeper.screen.home.presenter


import android.support.annotation.StringRes

import com.github.abhijitpparate.keeper.data.database.Note
import com.github.abhijitpparate.keeper.data.database.Profile

interface HomeContract {

    enum class NoteType {
        TEXT,
        IMAGE,
        VIDEO,
        AUDIO,
        DRAWING,
        LIST
    }

    interface View {
        fun setUserInfo(profile: Profile)
        fun setNotes(notes: List<Note>)

        fun showLoginScreen()
        fun showNewNoteScreen(noteType: NoteType)

        fun showNoteScreen(note: Note)

        fun setPresenter(presenter: Presenter)
        fun showProgressBar(bool: Boolean)
        fun makeToast(@StringRes strId: Int)
        fun makeToast(message: String)
    }

    interface Presenter {

        fun onRefresh()

        fun onLogoutClick()
        fun onNewNoteClick(noteType: NoteType)

        fun onNoteClick(note: Note)

        fun subscribe()
        fun unsubscribe()
    }
}
